### ISSUE TITLE
Deploy Progress Bars Not Showing

### ISSUE THREAD

**Issue by mcarvin8** [forcedotcom/cli#2276]
### Summary
After updating to the latest version of the `sf` CLI today (`@salesforce/cli/1.84.8 linux-x64 node-v18.16.1`), I noticed the deploy progress bars were not showing up during multiple deployments. The `SF_USE_PROGRESS_BAR` environment variable isn't currently set in our Docker image.

After downgrading to `sf` version `1.82.6` with no other changes, the deploy progress bars re-appeared.

### Steps To Reproduce

1. Upgrade to latest version of sf CLI (1.84.8)
2. Ensure the `SF_USE_PROGRESS_BAR` environment variable isn't set or is set to True.
3. Run a deployment - `sf project deploy start`
4. Confirm there are deploy progress bars.

### Expected result
Deploy progress bars are present from the start of the deployment til the end of the deployment.

### Actual result
No progress bars during the duration of the deployment with SF CLI version 1.84.8.

### System Information
Shell: Bash
Ubuntu Docker Image (latest)

```json
{
  "cliVersion": "@salesforce/cli/1.84.9",
  "architecture": "linux-x64",
  "nodeVersion": "node-v18.16.1",
  "osVersion": "Linux 5.15.108",
  "shell": "bash",
  "rootPath": "/usr/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.0 (core)",
    "@oclif/plugin-commands 2.2.16 (core)",
    "@oclif/plugin-help 5.2.10 (core)",
    "@oclif/plugin-not-found 2.3.26 (core)",
    "@oclif/plugin-plugins 3.1.3 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.19 (core)",
    "@oclif/plugin-version 1.3.5 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.39 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/cli 1.84.9 (core)",
    "apex 2.3.3 (core)",
    "auth 2.8.2 (core)",
    "data 2.3.26 (core)",
    "deploy-retrieve 1.13.3 (core)",
    "info 2.6.22 (core)",
    "limits 2.3.22 (core)",
    "login 1.2.13 (core)",
    "org 2.9.14 (core)",
    "schema 2.3.14 (core)",
    "settings 1.4.14 (core)",
    "sobject 0.1.26 (core)",
    "source 2.10.19 (core)",
    "telemetry 2.2.1 (core)",
    "templates 55.4.22 (core)",
    "trust 2.4.22 (core)",
    "user 2.3.17 (core)",
    "sfdx-git-delta 5.22.0 (user)"
  ]
}
```

**Comment by mcarvin8** [comment]
{
  "cliVersion": "@salesforce/cli/1.84.9",
  "architecture": "linux-x64",
  "nodeVersion": "node-v18.16.1",
  "osVersion": "Linux 5.15.108",
  "shell": "bash",
  "rootPath": "/usr/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.0 (core)",
    "@oclif/plugin-commands 2.2.16 (core)",
    "@oclif/plugin-help 5.2.10 (core)",
    "@oclif/plugin-not-found 2.3.26 (core)",
    "@oclif/plugin-plugins 3.1.3 (core)",
    "@oclif/plugin-search 0.0.17 (core)",
    "@oclif/plugin-update 3.1.19 (core)",
    "@oclif/plugin-version 1.3.5 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.39 (core)",
    "@oclif/plugin-which 2.2.22 (core)",
    "@salesforce/cli 1.84.9 (core)",
    "apex 2.3.3 (core)",
    "auth 2.8.2 (core)",
    "data 2.3.26 (core)",
    "deploy-retrieve 1.13.3 (core)",
    "info 2.6.22 (core)",
    "limits 2.3.22 (core)",
    "login 1.2.13 (core)",
    "org 2.9.14 (core)",
    "schema 2.3.14 (core)",
    "settings 1.4.14 (core)",
    "sobject 0.1.26 (core)",
    "source 2.10.19 (core)",
    "telemetry 2.2.1 (core)",
    "templates 55.4.22 (core)",
    "trust 2.4.22 (core)",
    "user 2.3.17 (core)",
    "sfdx-git-delta 5.22.0 (user)"
  ]
}

**Comment by shetzel** [comment]
I can't reproduce this.  Here are my steps:
1. `npm install -g @salesforce/cli@latest`
2. `sf -v`  --> @salesforce/cli/1.84.9 darwin-arm64 node-v18.16.0
3. `sf project deploy start`
<img width="1136" alt="Screenshot 2023-06-30 at 3 25 39 PM" src="https://github.com/forcedotcom/cli/assets/2703995/df215549-f159-4a2f-ba99-b1366035bbd3">

**Comment by shetzel** [comment]
There were some recent changes to this though, so maybe running in a docker container causes different TTY resolution with the changes.

The change to `plugin-deploy-retrieve` is here: https://github.com/salesforcecli/plugin-deploy-retrieve/pull/662/files
The change to `sf-plugins-core`: https://github.com/salesforcecli/sf-plugins-core/pull/324/files
The change to `oclif core`: https://github.com/oclif/core/pull/710/files

**Comment by mcarvin8** [comment]
Looks like this guy is running into a similar issue with GitLab pipeline (same CI platform I'm using)
https://github.com/forcedotcom/cli/issues/2249

At the end of my pipeline, I am parsing the JSON output to confirm the `result.Success` is True or False. But that happens at the end of the deployment. 

The progress bars are nice to show real-time progress while it's still running, especially in production pipelines that take longer than a sandbox. The JSON output also tends to over-flood the pipeline log with information that's harder for devs to interpret the results. So I just usually re-direct all that JSON output to a text file for parsing and print the success status back to the Pipeline log.

**Comment by gdman** [comment]
We are seeing the same thing - no progress bars in CI builds. In our case, Bitbucket Pipelines.

Progress bars have never worked amazingly well in non-tty environments but I agree with @PHI4133 - it's important to show real-time progress.

Not a blocker for us as at the moment as we're still using SFDX mdapi deploy commands but would be good to get this changed back.

**Comment by mcarvin8** [comment]
Confirming this is fixed in a GitLab CI/CD pipeline via sf version 2.1.7. Thanks!

![gitlab](https://github.com/forcedotcom/cli/assets/90224411/8f8208f8-c5d7-469e-90cd-c5c10c764159)



```
{
  "cliVersion": "@salesforce/cli/2.1.7",
  "architecture": "linux-x64",
  "nodeVersion": "node-v18.17.0",
  "osVersion": "Linux 5.15.108",
  "shell": "ash",
  "rootPath": "/usr/local/lib/node_modules/@salesforce/cli",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 2.3.2 (core)",
    "@oclif/plugin-commands 2.2.18 (core)",
    "@oclif/plugin-help 5.2.13 (core)",
    "@oclif/plugin-not-found 2.3.31 (core)",
    "@oclif/plugin-plugins 3.1.6 (core)",
    "@oclif/plugin-search 0.0.20 (core)",
    "@oclif/plugin-update 3.1.26 (core)",
    "@oclif/plugin-version 1.3.7 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.43 (core)",
    "@oclif/plugin-which 2.2.25 (core)",
    "@salesforce/cli 2.1.7 (core)",
    "apex 2.3.6 (core)",
    "auth 2.8.9 (core)",
    "data 2.5.1 (core)",
    "deploy-retrieve 1.16.0 (core)",
    "info 2.6.31 (core)",
    "limits 2.3.26 (core)",
    "login 1.2.20 (core)",
    "org 2.9.25 (core)",
    "schema 2.3.20 (core)",
    "settings 1.4.20 (core)",
    "sobject 0.1.37 (core)",
    "source 2.10.25 (core)",
    "telemetry 2.2.3 (core)",
    "templates 55.5.5 (core)",
    "trust 2.4.32 (core)",
    "user 2.3.24 (core)",
    "sfdx-git-delta 5.24.0 (user)"
  ]
}
```
